resources/ManageAPICredentials.php

```diff
@@ -0,0 +1,305 @@
+}
+/**
+ * Select API credentials for deletion.
+ *
+ * Find API credentials which have not been used for a given number of months
+ * and delete them or, if dry-run option is true, generate a summary report
+ * of the credentils found.
+ *
+ * @param bool              $dryRun             If true no action is taken and a report is generated instead
+ * @param \Doctrine\Orm\EntityManager $entitymanager A valid Doctrine Entity Manager
+ * @param \DateTime         $baseTime           Time from which interval of no-use is measured
+ * @param int               $deleteThreshold    The number of months of no-use which will trigger deletion
+ * @return array                                Credentials identified for deletion. Note: if dry-run is
+ *                                              NOT in operation all these credentials will have been deleted.
+ */
+function deleteCreds($dryRun, $entityManager, $baseTime, $deleteThreshold)
+{
+    $creds = getCreds($entityManager, $baseTime, $deleteThreshold);
+
+    if ($dryRun) {
+        reportDryRun($creds, "deleting");
+        return $creds;
+    }
+
+    $serv = new APIAuthenticationService();
+    $serv->setEntityManager($entityManager);
+
+    /* @var $api APIAuthentication */
+    foreach ($creds as $api) {
+        $serv->deleteAPIAuthentication($api);
+    }
+
+    return $creds;
+}
+/**
+ * Helper function to check if two API credentials have the same id.
+ *
+ * @return integer zero if equal, -1 if id1 < id2, 1 if id1 > id2
+ *
+*/
+function compareCredIds(APIAuthentication $cred1, APIAuthentication $cred2)
+{
+    $id1 = $cred1->getId();
+    $id2 = $cred2->getId();
+
+    if ($id1 == $id2) {
+        return 0;
+    };
+
+    return $id1 > $id2 ? 1 : -1;
+}
+/**
+ * Select API credentials for the sending of warning emails.
+ *
+ * Find API credentials which have not been used for a given number of months
+ * and send emails to the owners and site address, taken from the credential object,
+ * warning of impending deletion if the period of no-use reaches a given threshold.
+ * If dry-run option is true, generate a summary report of the credentials found
+ * instead of sending emails.
+ *
+ * @param bool          $dryRun             If true no action is taken and a report is generated instead
+ * @param \Doctrine\Orm\EntityManager $entitymanager A valid Doctrine Entity Manager
+ * @param \DateTime     $baseTime           Time from which interval of no-use is measured
+ * @param int           $warningThreshold   The number of months of no-use which triggers warning emails
+ * @param int           $deleteThreshold    The number of months of no-use which will trigger deletion
+ * @param string        $fromEmail          Email address to use as sender's (From:) address
+ * @param string        $replyToEmail       Email address for replies (Reply-To:)
+ * @param array         $deletedCreds       An array of credentials which have been selected for deletion
```

Comment:
maybe `$deletedCreds` would be better documented as $ignoreCreds?

Reply:
Probably, but as suggested in comment above, to use a single list from which deleted (even in dry-run) creds are removed is cleaner.